### PR TITLE
[FLINK-3482] implement union translation

### PR DESCRIPTION
**Pull request by vasia**
This PR adds union support to Table on Calcite. I've implemented a custom `JoinUnionTransposeRule`
because Calcite's only matches with LogicalUnion.

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/rules/logical/FlinkJoinUnionTransposeRule.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import org.apache.calcite.plan.RelOptRuleCall
+import org.apache.calcite.rel.RelNode
+import java.util.ArrayList
+import scala.collection.JavaConversions._
+import org.apache.calcite.rel.logical.LogicalJoin
+import org.apache.calcite.rel.logical.LogicalUnion
+import org.apache.calcite.rel.core.Join
+import org.apache.calcite.rel.core.Union
+
+class FlinkJoinUnionTransposeRule(
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/rules/logical/FlinkJoinUnionTransposeRule.scala]: Can you add a notice that this file was copied from Apache Calcite and briefly describe what you changed and why?

**Comment by fhueske** [comment]
Looks good. Can be merged after the comment was added. Thanks!

**Comment by fhueske** [comment]
+1 to merge

**Comment by vasia** [comment]
Thanks, I will merge.

**Pull request closed without merging**